OmgvaPOS/src/UserManagement/Service/UserService.cs

```diff
@@ -0,0 +1,127 @@
+using OmgvaPOS.ScheduleManagement.Models;
+using OmgvaPOS.OrderManagement.Models;
+using OmgvaPOS.UserManagement.Models;
+using OmgvaPOS.AuthManagement.DTOs;
+using OmgvaPOS.AuthManagement.Repository;
+
+namespace OmgvaPOS.UserManagement.Service
+{
+    public class UserService(IUserRepository userRepository, IAuthenticationRepository authenticationRepository) : IUserService
+    {
+        private readonly IUserRepository _userRepository = userRepository;
+        private readonly IAuthenticationRepository _authenticationRepository = authenticationRepository;
+        public UserResponse CreateUser(SignUpRequest userRequest)
+        {
+            if (userRequest == null)
+                throw new ArgumentNullException(nameof(userRequest));
+
+            User user = new()
+            {
+                Name = userRequest.Name,
+                Username = userRequest.Username,
+                Email = userRequest.Email,
+                Role = userRequest.Role,
+                Password = BCrypt.Net.BCrypt.EnhancedHashPassword(userRequest.Password, 13),
+                BusinessId = userRequest.BusinessId,
+                HasLeft = false
+            };
+
+            var createdUser = _authenticationRepository.SignUpUser(user);
+
+            if (createdUser == null)
+                throw new ArgumentNullException(nameof(userRequest));
+
+            UserResponse userResponse = new()
+            {
+                Id = createdUser.Id,
+                BusinessId = createdUser.BusinessId,
+                Name = createdUser.Name,
+                Username = createdUser.Username,
+                Email = createdUser.Email,
+                Role = createdUser.Role,
+                HasLeft = createdUser.HasLeft
+            };
+
+            return userResponse;
+        }
+        public List<UserResponse> GetAllUsers()
+        {
+            var users = _userRepository.GetUsers();
+            var usersResponse = new List<UserResponse>();
+            foreach (var user in users)
+            {
+                UserResponse userResponse = new()
+                {
+                    Id = user.Id,
+                    BusinessId = user.BusinessId,
+                    Name = user.Name,
+                    Username = user.Username,
+                    Email = user.Email,
+                    Role = user.Role,
+                    HasLeft = user.HasLeft
+                };
+                usersResponse.Add(userResponse);
+            }
+            return usersResponse;
+        }
+        public UserResponse GetUser(long id)
+        {
+            var user = _userRepository.GetUser(id);
+            UserResponse userResponse = new()
+            {
+                Id = user.Id,
+                BusinessId = user.BusinessId,
+                Name = user.Name,
+                Username = user.Username,
+                Email = user.Email,
+                Role = user.Role,
+                HasLeft = user.HasLeft
+            };
```

Comment:
Please create a separate Mapper class or at least a separate method in this class for mapping between DTO and Entity.  
In this UserService mapping from Entity -> UserResponse is done 4 times and makes service methods unnecessary long (>50% of code is just mapping logic; in this one, basically the whole method is mapping, we should extract it)